packages/smooth_app/lib/data_models/food_preferences_model.dart

```diff
@@ -0,0 +1,209 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:flutter/cupertino.dart';
+import 'package:smooth_app/database/user_database.dart';
+import 'package:smooth_app/temp/user_preferences.dart';
+import 'package:http/http.dart' as http;
+
+import 'dart:developer';
+
+/* preferences JSON sample:
+
+https://world.openfoodfacts.org/api/v0/preferences
+
+[
+{
+id: "not_important",
+name: "Not important"
+},
+{
+id: "important",
+name: "Important"
+},
+{
+id: "very_important",
+name: "Very important"
+},
+{
+id: "mandatory",
+name: "Mandatory"
+}
+]
+
+*/
+
+Future<List<Preference>> fetchPreferences(http.Client client) async {
```

Comment:
I'm not a big fan of method in the middle of nowhere (I prefer `static` methods, typically it would fit into `Preference`), but I don't know all the best practices in flutter.